### PR TITLE
[TS] LPS-173562 Adding Friendly URL redirect configuration for locale.prepend.friendly.url.style=2

### PR DESCRIPTION
**Pull request by jorgediaz-lr**
https://issues.liferay.com/browse/LPS-173562

Similar solution than PR https://github.com/liferay-lima/liferay-portal/pull/3965 but we have extracted **I18nFilter** to a module following the instructions of Brian here: https://github.com/brianchandotcom/liferay-portal/pull/129993#issuecomment-1424492524

**Commits:**

- 5b759cbc0e8df60c768fdb3f29ef9aa0f8ec3178 - remove dependencies of I18nFilter with MainServlet.java and PortalImpl.java removing all usages of `I18nFilter.getLanguageIds()` and `I18nFilter.setLanguageIds()` methods and use `I18nServlet` instead. The main difference between the languageIds stored information of I18nFilter and I18nServlet is first one stores the languageId code as is (_en_US_) and second one stores the languageId prefix (_/en_US_).
- 9cf1599a86df652cc30c41679e83ac8b833b7329 - move I18nFilter to a new module
- 7f7684c2bf1bb13c6acc1fc1f2ca20fc6966e650 - fix existing test, getting the service instead of creating a new I18nFilter object
- 5398f465d6e2948797a9d94092846201500e9172 - same code than previous PR: https://github.com/liferay-lima/liferay-portal/pull/3965
- 73c59863c629a2c1b2afcf0a1460ce23466aef5c - use helper to avoid repeating code
- 41832575c232d5c2e72a3448342564a582d7e40b - change description: remove reference to locale.prepend.friendly.url.style=2 because now it applies to any configuration
- 549954d6bf4910a394b9f99152678a0964ba18fd - buildLang
- 977bdbc5763c63307061e45a74f5233240315d63 - semver

/cc @danielmijarra @robertoDiaz

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/portal/portal-servlet-test/src/testIntegration/java/com/liferay/portal/servlet/filters/i18n/test/I18nFilterTest.java [hunk 1]
```diff
@@ -233,6 +241,27 @@ public void testGuestSpanishVirtualHostWithoutSessionCookieAlgorithm3()
...
+			serviceReferences.toString(), 1, serviceReferences.size());
+
+		Iterator<ServiceReference<Filter>> iterator =
+			serviceReferences.iterator();
+
+		ServiceReference<Filter> serviceReference = iterator.next();
+
+		return bundleContext.getService(serviceReference);
+	}
+
```
[the remark this answers is not in the document]
Comment (jorgediaz-lr) [reply modules/apps/portal/portal-servlet-test/src/testIntegration/java/com/liferay/portal/servlet/filters/i18n/test/I18nFilterTest.java]: Hi @adolfopa 

Yes, you are right, I copied the code from other tests that didn't use the Inject.

I have changed the commit, see new 7f7684c2bf1bb13c6acc1fc1f2ca20fc6966e650

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Comment by jorgediaz-lr** [comment]
ci:test:relevant

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/configuration/helper/FriendlyURLRedirectionConfigurationHelperImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ */
+@Component(
+	configurationPid = "com.liferay.friendly.url.internal.configuration.FriendlyURLRedirectionConfiguration",
+	service = FriendlyURLRedirectionConfigurationHelper.class
+)
+public class FriendlyURLRedirectionConfigurationHelperImpl
+	implements FriendlyURLRedirectionConfigurationHelper {
+
+	@Override
+	public String redirectionType(long companyId) {
```
Comment (adolfopa) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/configuration/helper/FriendlyURLRedirectionConfigurationHelperImpl.java:35]: Sorry I didn't see this before, but shouldn't this be `getRedirectionType`?

**Review by adolfopa — CHANGES_REQUESTED** [review]
@jorgediaz-lr Just a small naming issue; sorry I didn't see that in the previous review.

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/configuration/helper/FriendlyURLRedirectionConfigurationHelperImpl.java (hunk 2 above)
Comment (jorgediaz-lr) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/configuration/helper/FriendlyURLRedirectionConfigurationHelperImpl.java:35]: Yes, no problem

Comment (jorgediaz-lr) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/configuration/helper/FriendlyURLRedirectionConfigurationHelperImpl.java]: @adolfopa I have added a new commit with the change, see: 2c8d3fff9b82c34d54083e3030594358ed451912

**Comment by jorgediaz-lr** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by austinchiang** [comment]
ci:forward

**Comment by jorgediaz-lr** [comment]
ci:forward

**Comment by jorgediaz-lr** [comment]
ci:forward

**Comment by jorgediaz-lr** [comment]
ci:forward

**Comment by Jeremy-Ch** [comment]
The failures are not related to this pr, send to BChan manually. https://github.com/brianchandotcom/liferay-portal/pull/131955

**Pull request closed without merging**